esp32-wroom-rp/src/protocol.rs

```diff
@@ -43,14 +43,17 @@ pub(crate) enum NinaCommand {
     SendDataTcp = 0x44,
 }
 
-pub(crate) trait NinaConcreteParam {
+pub(crate) trait NinaConcreteParam
+where
+    Self: core::marker::Sized,
```

Comment:
Otherwise `Self` won't have a known size and you get this error:

```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> esp32-wroom-rp/src/protocol.rs:58:27
    |
58  |     fn new(data: &str) -> Result<Self, Error>;
    |                           ^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
note: required by a bound in `Result`
   --> /Users/jhodapp/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/result.rs:504:17
    |
504 | pub enum Result<T, E> {
    |                 ^ required by this bound in `Result`
help: consider further restricting `Self`
    |
58  |     fn new(data: &str) -> Result<Self, Error> where Self: Sized;
    |                                               +++++++++++++++++
```